lib/websocket/connection.js

```diff
@@ -357,9 +386,18 @@ class WebsocketFrame {
  */
 function receiveData (ws) {
   const { [kResponse]: response } = ws
-
+  let buffer = Buffer.alloc(0)
   response.socket.on('data', (chunk) => {
-    const frame = WebsocketFrame.from(chunk)
+    buffer = Buffer.concat([buffer, chunk])
```

Comment:
If we can get the payload length from the first chunk sent to us, it should be possible to allocate the buffer of that size once we receive the length, and then use `buffer.set` to "concatenate" them as we receive.

Reply:
makes sense. can we have a case where we get a chunk with the payload length still not in?

Reply:
I've added a TODO and I'll work on it soon, do you think we can merge this for now?

Reply:
btw, do you think it's a good idea to start working on some tests or is it still too early?

Reply:
If any of the tests pass from https://github.com/web-platform-tests/wpt/tree/master/websockets, we can add them. We don't necessarily need to add our own, but it won't harm anyone if you did. 👍

Reply:
Actually I doubt any of those tests will work until we can setup a WebSocket server.